LOKI2R.GCL.R

```diff
@@ -52,132 +84,126 @@ LOKI2R.GCL=function(sillyvec,username,password){
+    drv <- RJDBC::JDBC("oracle.jdbc.OracleDriver", classPath=path.expand("~/R/ojdbc8.jar"), " ")
     
   }
   
-  url <-LOKI_URL.GCL()
+  url <- LOKI_URL.GCL() #This is a function that gets the correct URL to access the database on the oracle cloud
   
-  con <- dbConnect(drv,url=url,user=username,password=password)
+  con <- RJDBC::dbConnect(drv, url = url, user = username, password = password) #The database connection
   
   loci <- LocusControl$locusnames
-
+  
   nloci <- length(loci)
-
+  
   ploidy <- LocusControl$ploidy
-
+  
   alleles <- LocusControl$alleles
-    
+  
   nalleles <- LocusControl$nalleles 
-
-  gnoqry <- paste("SELECT * FROM AKFINADM.V_GNOQRY WHERE LOCUS IN (",paste0("'",loci,"'",collapse=","),") AND SILLY_CODE IN (",paste0("'",sillyvec,"'",collapse=","),")",sep="")
-
-  dataAll0 <- dbGetQuery(con,gnoqry)   
-
-  dataAllbool <- dataAll0$PK_TISSUE_TYPE==dataAll0$PREFERRED_TISSUE 
-
+  
+  gnoqry <- paste("SELECT * FROM AKFINADM.V_GNOQRY WHERE LOCUS IN (", paste0("'", loci, "'", collapse = ","), ") AND SILLY_CODE IN (", paste0("'", sillyvec, "'", collapse = ","), ")", sep = "") #Gentoype query
+  
+  dataAll0 <- RJDBC::dbGetQuery(con, gnoqry)  #Pulling data from LOKI using the connection and genotype query
+  
+  dataAllbool <- dataAll0$PK_TISSUE_TYPE == dataAll0$PREFERRED_TISSUE 
+  
   bothNAbool <- is.na(dataAll0$PREFERRED_TISSUE) &  is.na(dataAll0$PK_TISSUE_TYPE)
   
   dataAllbool[is.na(dataAllbool)] <- FALSE 
-
+  
   dataAllbool[bothNAbool] <- TRUE  
   
-  dataAll <- dataAll0[dataAllbool,]
-
-  attrCol <- c("TEST_TYPE","LOCUS","ALLELE_1","ALLELE_2","SUITE_NAME","PREFERRED_TISSUE") 
+  dataAll <- dataAll0[dataAllbool, ] %>% 
+    tibble::as_tibble() %>% 
+    dplyr::select(-PREFERRED_TISSUE, -TEST_TYPE)
+  
+  discon <- RJDBC::dbDisconnect(con)
+  
+  #Create tibble data fame for each silly in sillyvec
+  for(silly in sillyvec){ 
     
-  attnames <- names(dataAll)[!names(dataAll)%in%attrCol]
+    message0 <- paste0(silly, ".gcl created ", match(silly, sillyvec)," of ", length(sillyvec)," completed.") 
     
-  discon <- dbDisconnect(con)
+    sillydata <- dataAll %>% 
+      dplyr::filter(SILLY_CODE==silly)
+    
+    ids <- sillydata$FISH_ID %>% 
+      unique() %>% 
+      sort() %>% 
+      as.character()
+    
+    sillyvials <- paste(silly, ids, sep = "_")
     
-  for(silly in sillyvec){ 
-
-    message0 <- paste0(silly,".gcl created ", match(silly,sillyvec)," of ",length(sillyvec)," completed.") 
-  
-    sillydata <- subset(dataAll,SILLY_CODE==silly)
-   
-    ids <- as.character(sort(unique(sillydata$FISH_ID)))   
-      
-    sillyvials <- paste(silly,ids,sep="_")
-      
     nind <- length(sillyvials)
-
+    
     if(length(ids) == 0){
-
+      
       message0 <- paste0(silly," is empty.")
-
+      
       message(message0)
-
+      
       next
-
-    }
       
-    scores <- array(data=NA,dim=c(nind,length(loci),max(ploidy)),dimnames=list(ids,loci,paste0("Dose",seq(max(ploidy)))))

```

Comment:
Hey Andy, I definitely dig the new tibble format for .gcl objects, but one thing I did notice since we are no longer filling in a pre-defined array, sillys that don't have data for certain loci no longer have for NA's for those loci in the final tibble, they just don't have those loci (columns) at all in the final tibble. Currently there is no way that a user would know that the collection didn't have genotypes for all loci until you try `RemoveNAindiv.GCL`, and you'd only know then because the function bombs because it can't find those loci to see if there are NAs. Make sense? 

How should we deal with this? Based on our current workflows, I think that each silly should probably have columns for all loci in LocusControl and fill in NAs for any loci without genotypes, but I'm open to other ideas.

Reply:
We could also toss out a message that says that loci are missing for some collections and automatically dump out a tibble showing which sillys are missing which loci.

Reply:
I think this is a good idea.  That way you know right away if some of the sillys your are trying to pull have missing data.  Sillys without any data will not get pulled from LOKI at all because they don't exist in the lookup table.  I can add those to the missing data tibble as well.